### DISCUSSION TITLE
What do terms in parentheses mean in enums in Mint ?

### DISCUSSION THREAD

**Discussion by serge-hulne** [mint-lang/mint#562] [Q&A]
If I understand correctly, enums allow defining aggregate types, each one stipulating a possible concrete type a variable could take.

So, I would  expect a enum to look something like:

```
enum DayOfWeek {
    String
    Int
}
```

But what does the additional information in parentheses mean ?

```
enum UserState {
  LoggedIn(User)
  Visitor
}
```

**Comment by Sija** [comment]
See the [documentation](https://mint-lang.com/guide/reference/enums).

**Reply by serge-hulne** [reply]
I did. It does not answer my question.

**Reply by gdotdesign** [reply]
> But what does the additional information in parentheses mean ?

It means that that variant `LoggedIn` can hold another value, in this case a `User`. A good way to think about these kinds of variants as boxes, in this case you put a `User` in the box which later on you can access.
